components/script/dom/webglframebuffer.rs

```diff
@@ -289,12 +290,37 @@ impl WebGLFramebuffer {
             };
 
             if matched {
-                *attachment.borrow_mut() = None;
-                self.update_status();
+                closure(attachment);
             }
         }
     }
 
+    pub fn detach_renderbuffer(&self, rb: &WebGLRenderbuffer) {
+        self.call_on_matching_renderbuffers(rb, |att| {
+            *att.borrow_mut() = None;
+            self.update_status();
```

Comment:
I think with the `FnMut` clause we can track here and below whether at least one has been matched and update the status only once?